### DISCUSSION TITLE
Asking Information about automations for Keykcloak

### DISCUSSION THREAD

**Discussion by AngraPatrick** [keycloak/keycloak#10435] [Q&A]
Hi dears

My name is Samir and this account is my company account.
With this post I want ask you if there are some available software for make automation in Keycloak.

I know this software ( Keycloak ) was made in java so I thought could be exist some tools written in java ( .jar ) for reach the PostgreSQL database using an API. I am asking you that because if possible I prefer to don't use the mouse on interface, the interface itself I think it might decrease my productivity and also I would like to make some automatic change on groups and roles.

Actually my company is testing this software using docker compose with some modifications ( for Nginx, Frontend client, ... ), our file is based on this: https://github.com/keycloak/keycloak-containers/blob/main/docker-compose-examples/keycloak-postgres.yml

If Jboss is the solution where I can find necessary documentation connected to Keycloak?

Best Regards
Samir

**Comment by Wolfsrudel** [comment]
See: https://docs.ansible.com/ansible/latest/collections/community/general/keycloak_realm_module.html
and: https://docs.ansible.com/ansible/latest/collections/community/general/keycloak_client_module.html
and: https://duckduckgo.com/?q=ansible+keycloak

**Reply by AngraPatrick** [reply]
Oh Yes I know Ansible, I never thought it as possible solution for this problem in keycloak, this because in past we ( company ) were used to use it for manage a cluster network, thank you a ton.
